### ISSUE TITLE
[🐞] How is it possible to handle `qwikCity` in Library Mode?

### ISSUE THREAD

**Issue by saeedhbi** [BuilderIO/qwik#4920]
### Which component is affected?

Qwik City (routing)

### Describe the bug

I don't know if this is a bug or something else, but based on my needs, since I'm using my custom pnpm monorepo workspace, the problem is serving `@qwik-city-plan` in packages which are using `qwik-city` package. It seems it's only can be served from the root of the builder app and not the library mode packages.

Currently, I have multiple packages (libraries) that are using some `qwik-city` exports like `useLocation` or `server$` or etc. So, there are two ways to build these packages:

1. Building with vite `qwikCity()` plugin which will build in different ways of libraries (output is not a single export)
2. Make external everything regarding `qwikCity`

I don't want to use the first one because in that way, the way of communication between packages will be changed but in the case of using the second way, I don't know the way of resolving qwik modules because at the end of the building, the vite config of root application is responsible but the package is installed in the root of the repo.

Is there anything that I'm missing? Should I transfrom `qwikCity` in my custom way for every package separately? I couldn't find anything regarding this. Thanks for your helps.

### Reproduction

The is no reproduction url!

### System Info

```shell
System:
    OS: Linux 5.15 elementary OS 6.1 Jólnir
    CPU: (4) x64 Intel(R) Core(TM) i7-4510U CPU @ 2.00GHz
    Memory: 3.11 GB / 15.52 GB
    Container: Yes
    Shell: 5.8 - /usr/bin/zsh
  Binaries:
    Node: 20.1.0 - ~/.nvm/versions/node/v20.1.0/bin/node
    npm: 9.6.4 - ~/.nvm/versions/node/v20.1.0/bin/npm
  Browsers:
    Brave Browser: 100.1.37.113
    Chrome: 114.0.5735.198
  npmPackages:
    vite: 4.3.5 => 4.3.5
```

**Comment by juanpmarin** [comment]
@saeedhbi did you find a solution for using qwik city in libraries?

**Comment by saeedhbi** [comment]
> @saeedhbi did you find a solution for using Qwik City in libraries?

No. Since the responsibility of QwikCity is serving to route, and the responsibility of libraries is serving logic, I just mapped the Qwik City only on application rather than libraries to the building process. In libraries, I ignored building them and passed them to external libraries. On applications, the building process will handle it properly.
